Chess/src/Bishop.cpp

```diff
@@ -0,0 +1,79 @@
+#include "../include/Bishop.h"
+#include <vector>
+#include <utility>
+
+using namespace std;
+
+Bishop::Bishop()=default;
+
+Bishop::Bishop(char color, std::pair<int, int> Cord,Chess * chessObj) : Piece(color, Cord,chessObj) {}
+
+std::pair<int, int> Bishop::PiecePlace() const {
+    return this->PieceCords;
+}
+
+std::vector<std::pair<int, int>> Bishop::PieceMoves() const {

```

Comment:
Why do you need to calculate all the possible moves instead of checking that the one move that the player is trying to play is valid?